### PR TITLE
[api] Move text wrapping handling logic from layout table code to QgsTextRenderer

### PR DESCRIPTION
**Pull request by nyalldawson**
Allows other users of QgsTextRenderer to take advantage of the automatic line wrapping behaviour

~~(This is well covered by the layout table tests, but I'd like to add some specific unit tests for QgsTextRenderer before merging)~~

### CODE REVIEW

**Comment by nyalldawson** [comment]
Supersedes https://github.com/qgis/QGIS/pull/41664

**Comment by nyalldawson** [comment]
Additional tests have been added

**Review by nirvn — APPROVED** [review]
Can't see anything wrong with the ported code/logic. One formatting question.

File: src/core/layout/qgslayouttable.cpp [hunk 1]
```diff
@@ -1189,16 +1180,16 @@ bool QgsLayoutTable::calculateMaxRowHeights()
...
     else
     {
-      heights[i] = QgsTextRenderer::textHeight( context, cellFormat, QStringList() << col.heading(), QgsTextRenderer::Rect ) / context.convertToPainterUnits( 1, QgsUnitTypes::RenderMillimeters )
+      heights[i] = QgsTextRenderer::textHeight( context,
+                   cellFormat,
+                   QStringList() << col.heading(), QgsTextRenderer::Rect,
+                   nullptr,
+                   mWrapBehavior == WrapText ? Qgis::TextRendererFlag::WrapLines : Qgis::TextRendererFlags(),
+                   context.convertToPainterUnits( mColumns.at( i ).width(), QgsUnitTypes::RenderMillimeters )
+                                              )
```
[the remark this answers is not in the document]
Comment (nyalldawson) [reply src/core/layout/qgslayouttable.cpp:1191]: It is autoformatted, and keeps jumping back to this. I think astyle is getting confused with the <<

[the remark this answers is not in the document]
Comment (nirvn) [reply src/core/layout/qgslayouttable.cpp:1191]: OK, well, then there's nothing we can do about it :) merge at will.

**Comment by nyalldawson** [comment]
Thanks @nirvn!

**Pull request merged**